enterprise/frontend/src/embedding-sdk/components/public/InteractiveDashboard/InteractiveAdHocQuestion.tsx

```diff
@@ -0,0 +1,82 @@
+import { useEffect, useState } from "react";
+
+import type { SdkClickActionPluginsConfig } from "embedding-sdk";
+import { InteractiveQuestionResult } from "embedding-sdk/components/public/InteractiveQuestion/InteractiveQuestionResult";
+import { useDispatch, useSelector } from "metabase/lib/redux";
+import { initializeQBRaw } from "metabase/query_builder/actions";
+import { getQueryResults } from "metabase/query_builder/selectors";
+
+interface InteractiveAdHocQuestionProps {
+  questionPath: string;
+  onNavigateBack: () => void;
+
+  withTitle?: boolean;
+  height?: number;
+  plugins?: SdkClickActionPluginsConfig;
+}
+
+export const InteractiveAdHocQuestion = ({
+  questionPath,
+  onNavigateBack,
+  withTitle = true,
+  height,
+  plugins,
+}: InteractiveAdHocQuestionProps) => {
+  const dispatch = useDispatch();
+
+  const queryResults = useSelector(getQueryResults);
+
+  const [isQuestionLoading, setIsQuestionLoading] = useState(true);
+
+  const loadQuestion = async (
+    dispatch: ReturnType<typeof useDispatch>,
+    questionUrl: string,
+  ) => {
+    setIsQuestionLoading(true);
+
+    const { location, params } = getQuestionParameters(questionUrl);
+    try {
+      await dispatch(initializeQBRaw(location, params));
+    } catch (e) {
+      console.error(`Failed to get question`, e);
+      setIsQuestionLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    loadQuestion(dispatch, questionPath);
+  }, [dispatch, questionPath]);
+
+  useEffect(() => {
+    if (queryResults) {
+      setIsQuestionLoading(false);
+    }
+  }, [queryResults]);
```

Comment:
we _might_ need to watch out for race conditions here. might be a good idea to test what happens when multiple question paths are loaded consequently. I think this should be fine, but it would be good to test this.